runtime/src/bank.rs

```diff
@@ -390,29 +391,38 @@ impl CachedExecutors {
             entry
         } else {
             saturating_add_assign!(self.stats.misses, 1);
-            if self.executors.len() >= self.max {
-                let mut least = u64::MAX;
-                let default_key = Pubkey::default();
-                let mut least_key = &default_key;
-
-                for (key, entry) in self.executors.iter() {
+            let mut counts = self
+                .executors
+                .iter()
+                .map(|(key, entry)| {
                     let count = entry.prev_epoch_count + entry.epoch_count.load(Relaxed);
-                    if count < least {
-                        least = count;
-                        least_key = key;
-                    }
-                }
-                let least_key = *least_key;
+                    (key, count)
+                })
+                .collect::<Vec<_>>();
+            counts.sort_unstable_by_key(|(_, count)| *count);
+
+            let primer_count = Self::get_primer_count(counts.as_slice());
+
+            let maybe_least_key = if self.executors.len() >= self.max {
+                Some(counts.last().map(|least| *least.0).unwrap_or_default())
+            } else {
+                None
+            };
+
+            drop(counts);
+
+            if let Some(least_key) = maybe_least_key {
                 let _ = self.executors.remove(&least_key);
                 self.stats
                     .evictions
                     .entry(least_key)
                     .and_modify(|c| saturating_add_assign!(*c, 1))
                     .or_insert(1);
             }
```

Comment:
```suggestion
            if self.executors.len() >= self.max {
                if let Some(least_key) = counts.last().map(|least| *least.0) {
                    let _ = self.executors.remove(&least_key);
                    self.stats
                        .evictions
                        .entry(least_key)
                        .and_modify(|c| saturating_add_assign!(*c, 1))
                        .or_insert(1);
                }
            }
```

Reply:
I had it like this initially, but you can't take a `mut` ref on `self.executors` here w/o the `drop` (or `clone`ing all the pubkeys).

I'll add a comment so that's more apparent though

Reply:
I compiled this code just fine

Reply:
You're right!  I must not have been copying `least_key`.  Thanks!